src/envs/coffee.py

```diff
@@ -0,0 +1,756 @@
+    machine_y: ClassVar[float] = y_ub - machine_y_len - init_padding
+    button_x: ClassVar[float] = machine_x + machine_x_len / 2
+    button_y: ClassVar[float] = machine_y
+    button_z: ClassVar[float] = 3 * machine_z_len / 4
+    button_radius: ClassVar[float] = 0.2 * machine_x_len
+    # Jug settings.
+    jug_radius: ClassVar[float] = (0.8 * machine_x_len) / 2.0
+    jug_height: ClassVar[float] = 0.15 * (z_ub - z_lb)
+    jug_init_x_lb: ClassVar[float] = machine_x - machine_x_len + init_padding
+    jug_init_x_ub: ClassVar[float] = machine_x + machine_x_len - init_padding
+    jug_init_y_lb: ClassVar[float] = y_lb + jug_radius + init_padding
+    jug_init_y_ub: ClassVar[
+        float] = machine_y - machine_y_len - jug_radius - init_padding
+    jug_handle_x_offset: ClassVar[float] = 0.0
+    jug_handle_y_offset: ClassVar[float] = -(1.05 * jug_radius)
+    jug_handle_height: ClassVar[float] = 3 * jug_height / 4
+    # Dispense area settings.
+    dispense_area_x: ClassVar[float] = machine_x + machine_x_len / 2
+    dispense_area_y: ClassVar[float] = machine_y - 1.1 * jug_radius
+    # Cup settings.
+    cup_radius: ClassVar[float] = 0.6 * jug_radius
+    cup_init_x_lb: ClassVar[float] = x_lb + cup_radius + init_padding
+    cup_init_x_ub: ClassVar[
+        float] = machine_x - machine_x_len - cup_radius - init_padding
+    cup_init_y_lb: ClassVar[float] = jug_init_y_lb
+    cup_init_y_ub: ClassVar[float] = jug_init_y_ub
+    cup_capacity_lb: ClassVar[float] = 0.075 * (z_ub - z_lb)
+    cup_capacity_ub: ClassVar[float] = 0.15 * (z_ub - z_lb)
+    cup_target_frac: ClassVar[float] = 0.75  # fraction of the capacity
+    # Simulation settings.
+    pour_x_offset: ClassVar[float] = 1.5 * (cup_radius + jug_radius)
+    pour_y_offset: ClassVar[float] = cup_radius
+    pour_z_offset: ClassVar[float] = 1.1 * (cup_capacity_ub + jug_height - \
+                                            jug_handle_height)
+    pour_velocity: ClassVar[float] = cup_capacity_ub / 10.0
+    max_position_vel: ClassVar[float] = 0.5
+    max_angular_vel: ClassVar[float] = tilt_ub
+    max_finger_vel: ClassVar[float] = 1.0
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Types
+        self._robot_type = Type("robot", ["x", "y", "z", "tilt", "fingers"])
+        self._jug_type = Type("jug", ["x", "y", "is_held", "is_filled"])
+        self._machine_type = Type("machine", ["is_on"])
+        self._cup_type = Type(
+            "cup",
+            ["x", "y", "capacity_liquid", "target_liquid", "current_liquid"])
+
+        # Predicates
+        self._CupFilled = Predicate("CupFilled", [self._cup_type],
+                                    self._CupFilled_holds)
+        self._Holding = Predicate("Holding",
+                                  [self._robot_type, self._jug_type],
+                                  self._Holding_holds)
+        self._InMachine = Predicate("InMachine",
```

Comment:
kind of weird that this isn't called `JugInMachine` when the other one is called `JugFilled`